pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -288,43 +295,57 @@ public Response updateFunction(final String tenant, final String namespace, fina
     }
 
     public Response deregisterFunction(final String tenant, final String namespace, final String functionName,
-            String clientRole) {
+            String sourceName, String sinkName, String clientRole) {
```

Comment:
For 
```
public Response deregisterFunction(final String tenant, final String namespace, final String functionName, String sourceName, String sinkName, String clientRole)
```
I think it might be simpler and clearer to just have something like:
```
public Response deregisterFunction(final String tenant, final String namespace, final String name, String/enum type, String clientRole)
```

There is already constants declared i.e.

```
    private static final String FUNCTION = "Function";
    private static final String SOURCE = "Source";
    private static final String SINK = "Sink";
```

Reply:
Thats a good idea! Will change it to use that format

Reply:
@jerrypeng I have converted.